docker-api/src/main/java/com/yahoo/vespa/hosted/dockerapi/ContainerResources.java

```diff
@@ -36,11 +41,20 @@
             throw new IllegalArgumentException("memoryBytes must be a positive integer or 0 for unlimited, was " + memoryBytes);
     }
 
+    /**
+     * Create container resources from required fields.
+     *
+     * @param cpus the amount of vcpu that should be exclusively available to this container. This is a hard limit:
+     *             More than this amlunt will never be available. To allow an unlimited amount use 0.
+     * @param cpuCores the amount of vcpu that should ideally be allocated to this container if there are no other
+     *                 constraints on resources. To allow an unlimited amount use 0.
+     * @param memoryGb the exact amount of memory that must be available to this container.
```

Comment:
This is the same as `cpus`: it's a hard upper bound for the container and not a guarantee to it.

Reply:
Same as the above. We actually guarantee it, and must do so always.